sympy/solvers/ode.py

```diff
@@ -277,10 +277,10 @@
 #: ``best``, and ``all_Integral`` meta-hints should not be included in this
 #: list, but ``_best`` and ``_Integral`` hints should be included.
 allhints = (
-    "separable",
-    "1st_exact",
     "1st_linear",
     "Bernoulli",
+    "separable",
+    "1st_exact",
```

Comment:
Tests will likely fail because of a change in the output format. You should investigate them and see how things compare. If the format is better, or dsolve is significantly faster, then we should make the change.